WhateverGreen/kern_igfx.cpp

```diff
@@ -503,7 +437,7 @@ bool IGFX::processKext(KernelPatcher &patcher, size_t index, mach_vm_address_t a
 		bool bklKabyFb = realFramebuffer == &kextIntelKBLFb && cflBacklightPatch == CoffeeBacklightPatch::On;
 		// Solve ReadRegister32 just once as it is shared
 		if (bklCoffeeFb || bklKabyFb ||
-			RPSControl.enabled || ForceWakeWorkaround.enabled || coreDisplayClockPatch) {
+			RPSControl.enabled || ForceWakeWorkaround.enabled || modCoreDisplayClockFix.enabled) {
```

Comment:
This is a bit ugly. Probably we want a separate bool for submodules like `requiresGenericRegisterIo`. We can make a joint version of this bool in IGFX, assign it in processKernel as an OR of all the submodules values, and then use here.

 For now since not all entries are submodules we can assign it some hardcoded entries too. But please add a FIXME.

Reply:
Totally agreed. At this moment, backlight fix is the only one that injects code to `readRegister32/writeRegister32`, while others only need them as helper functions. 

The worst case is that multiple fixes request to inject code to those functions, so we probably need something like `GenericRegisterModifier` to orchestrate injection requests. The same thing might happen in the AUX register case.

For the global controller access, we could apply the same concept as well. Apple has refactored a lot of code from `AppleIntelFramebufferController` to the new class `AppleIntelPort`, so a new fix might need the controller instance to fetch the framebuffer instance associated with a port via `AppleIntelFramebufferController::getFBFromPort()`.